src/lib/components/editor/CodeMirrorResponse.svelte

```diff
@@ -0,0 +1,114 @@
+<script lang="ts">
+  import { afterUpdate, onDestroy, onMount } from "svelte";
+  import {
+    RequestDataType,
+    ResponseFormatter,
+  } from "$lib/utils/enums/request.enum";
+  import { isHorizontal } from "$lib/store/request-response-section";
+  import { EditorView } from "codemirror";
+  import { javascript } from "@codemirror/lang-javascript";
+  import { baseTheme, basicSetup, jsonSetup } from "./codeMirrorTheme";
+  import { EditorState, Compartment } from "@codemirror/state";
+  import { html } from "@codemirror/lang-html";
+  import { xml } from "@codemirror/lang-xml";
+  export let currentTabId: string;
+  export let rawTab: RequestDataType;
+  export let rawValue: string;
+  export let formatter: string;
+  let selectedTabId = currentTabId;
+  const languageConf = new Compartment();
+  let codeMirrorEditorDiv: HTMLDivElement;
+  let codeMirrorView: EditorView;
+  function initalizeCodeMirrorEditor(value: string) {
+    let state = EditorState.create({
+      doc: value,
+      extensions: [
+        basicSetup,
+        baseTheme,
+        languageConf.of([]),
+        EditorState.readOnly.of(true),
+      ],
+    });
+    codeMirrorView = new EditorView({
+      parent: codeMirrorEditorDiv,
+      state: state,
+    });
+  }
+  onMount(() => {
+    initalizeCodeMirrorEditor(rawValue);
+  });
+  afterUpdate(() => {
+    if (selectedTabId !== currentTabId) {
```

Comment:
This can be converted into a function that takes codeMirrorView as an argument. We can use the same in both request and response.